test/framework/src/main/java/org/elasticsearch/test/AbstractQueryTestCase.java

```diff
@@ -83,15 +82,22 @@
     private static final int NUMBER_OF_TESTQUERIES = 20;
 
     public final QB createTestQueryBuilder() {
-        QB query = doCreateTestQueryBuilder();
-        //we should not set boost and query name for queries that don't parse it
+        QB query;
         if (supportsBoostAndQueryName()) {
-            if (randomBoolean()) {
-                query.boost(2.0f / randomIntBetween(1, 20));
-            }
-            if (randomBoolean()) {
-                query.queryName(createUniqueRandomName());
-            }
+            query = createTestQueryBuilder(true, true);
```

Comment:
May be instead  of `supportsBoostAndQueryName()` create two methods `supportsBoost()` and `supportsQueryName()` that others can override?

Reply:
Yes! I'm glad you ask for it, that's what I did first before reverting my change to no cause too much changes. Refactoring is back again in 8e9c70d1ebcf6a277024eed8efd8ed180c76c776 :+1: